server/cluster/unsafe_recovery_controller.go

```diff
@@ -281,53 +282,78 @@ func (u *unsafeRecoveryController) HandleStoreHeartbeat(heartbeat *pdpb.StoreHea
 		return
 	}
 
-	allCollected, err := u.collectReport(heartbeat)
-	if err != nil {
-		u.err = err
-		if u.handleErr() {
-			return
-		}
+	storeID := heartbeat.Stats.StoreId
+	if _, isFailedStore := u.failedStores[storeID]; isFailedStore {
```

Comment:
why move it out from `collectReport`

Reply:
Was thinking moving all the error handling inside this function to make it more clear, since the logic has changed again, reverting this.